src/m365/commands/docs.spec.ts

```diff
@@ -0,0 +1,83 @@
+import Command from '../../Command';
+import commands from './commands';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { Logger } from '../../cli/Logger';
+import * as open from 'open';
+import { telemetry } from '../../telemetry';
+import { pid } from '../../utils/pid';
+import { session } from '../../utils/session';
+import { sinonUtil } from '../../utils/sinonUtil';
+import { Cli } from '../../cli/Cli';
+const packageJSON = require('../../../package.json');
+const command: Command = require('./docs');
+
+describe(commands.DOCS, () => {
+  let log: any[];
+  let logger: Logger;
+  let cli: Cli;
+  let loggerLogSpy: sinon.SinonSpy;
+  let getSettingWithDefaultValueStub: sinon.SinonStub;
+  let openStub: sinon.SinonStub;
+
+  before(() => {
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+  });
+
+  beforeEach(() => {
+    log = [];
+    cli = Cli.getInstance();
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any)._open = open;
+    openStub = sinon.stub(command as any, '_open').resolves();
+    getSettingWithDefaultValueStub = sinon.stub(cli, 'getSettingWithDefaultValue').returns(false);
+  });
+
+  afterEach(() => {
+    loggerLogSpy.restore();
+    getSettingWithDefaultValueStub.restore();
+    openStub.restore();
+    sinon.restore();
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
+  });
```

Comment:
Maybe my last review was not so clear, sorry if that was the case.
But what I wanted to achieve is that we remove the `sinonUtil.restore` from the `after` hook and replace that with `sinon.restore()` (this is a new and better way how to restore the entire sandbox).

In the `afterEach` hook we can restore stubs like we used to, so you can remove `sinon.restore()` there. Also now you are restoring each stub, let's refactor that a bit to use `sinonUtil.restore` (like you currently do in the `after` hook.

Reply:
Are you suggesting, use sinon.restore() in the after hook to restore the entire sandbox. And, use sinonUtil.restore to restore the individual stubs in the afterEach hook.

like the below:

```
afterEach(() => {
    sinonUtil.restore([
      telemetry.trackEvent,
      pid.getProcessName,
      session.getId,
      loggerLogSpy,
      getSettingWithDefaultValueStub,
      openStub
    ]);
  });

  after(() => {
    sinon.restore();
  });
```

Reply:
Exactly!

Reply:
I initially used sinon.restore() in the afterEach hook because it should clean up all resources after each test case runs. We may not need to restore individual stubs. 

As per your above comment, I will change it as below:

```
afterEach(() => {
    loggerLogSpy.restore();
    getSettingWithDefaultValueStub.restore();
    openStub.restore();
  });

  after(() => {
    sinon.restore();
  });
```

Reply:
What I actually meant was that we use the `sinonUtil.restore([...])` function in the `afterEach` to clear these 3 stubs, and use `sinon.restore()` in the `after` hook. Could you update this? Sorry for being so unclear!

We can't just use `sinon.restore()` in the `afterEach` because that would also wipe the stubs we are defining in the `before` hook, which is not what we want.